File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        maxx = 0
        n = len(heights)
        l ,r = 0 , n-1
        while(l < r):
            h = min(heights[l] , heights[r])
            area = (r-l) * h
            maxx = max (maxx,area)

            if heights[l] < heights[r]:
                l+=1
            else:
                r-=1

        return maxx

        

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l = 0;
        int maxx = 0;
        int counts[26] = {0};
        for (int r = 0; r < s.length(); ++r){
            counts[s[r] - 'A']++;
            int curr = 0;
            for(int i = 0 ; i < 26; ++i){
                curr = max(curr,counts[i]);
            } 
            while(r-l+1 - curr > k){
                counts[s[l] - 'A']--;
                l++;
                curr = 0;
                for(int i = 0; i<26; ++i){
                    curr = max(curr,counts[i]);
                }
            }
            
            maxx = max(maxx , r-l+1);
        }

        return maxx;
    }
};
